src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs

```diff
@@ -101,17 +102,25 @@ private async Task UpdateRecentsList(NotifyCollectionChangedEventArgs args)
 
 				Empty.Visibility = Visibility.Collapsed;
 
-				switch (args.Action)
+				switch (e.Action)
 				{
 					// currently everything falls under Reset
 					default:
-						recentItemsCollection.Clear();
-						var recentFiles = App.RecentItemsManager.RecentFiles; // already sorted, add all in order
-						foreach (var recentFile in recentFiles)
 						{
-							await AddItemToRecentListAsync(recentFile);
+							var recentFiles = App.RecentItemsManager.RecentFiles; // already sorted, add all in order
+							foreach (RecentItem elem in recentItemsCollection.ToList())
+							{
+								if (!recentFiles.Any(x => x.Equals(elem)))
+								{
+									recentItemsCollection.Remove(elem);
+								}
+							}
+							for (int i = 0; i < recentFiles.Count; i++)
+							{
+								await AddItemToRecentListAsync(recentFiles[i], i);
+							}
+							break;
```

Comment:
Imprivement: slightly smarter update of recent items list. Do not clear the whole list every time.

Reply:
Seems a little expensive to do `recentFiles.Any` inside a loop, wonder if there is a better way we can do this.